src/lib/zkapp.ts

```diff
@@ -206,13 +208,16 @@ function wrapMethod(
             assertStatePrecondition(this);
             return result;
           } else {
-            // in a transaction, also add a lazy proof to the self party
-            // (if there's no other authorization set)
+            // called smart contract at the top level, in a transaction!
+            // => attach ours to the current list of parties
+            let party = context.selfUpdate;
+            if (!context.isCallback) {
+              Mina.currentTransaction()?.parties.push(party);
```

Comment:
important change! the `this.self` party isn't attached to the list of parties automatically anymore. so we have to do it explicitly here. (it was just annoying that parties that were intended to be child parties popped up on the top level as well)

Reply:
Why is `this.self` not being added anymore?